example/spec/parallel/lzardsBackup/LzardsBackupSpec.js

```diff
@@ -172,27 +176,61 @@ describe('The Lzards Backup Task ', () => {
       beforeAllFailed = true;
       throw error;
     }
-  });
+  };
 
-  it('succeeds', () => {
-    if (beforeAllFailed) fail('beforeAll() failed');
-    else {
-      expect(lzardsBackupOutput.FunctionError).toBe(undefined);
-    }
-  });
+  describe('The Lzards Backup Task with override configured', () => {
+    it('invokes successfully', async () => {
+      const taskConfig = {
+        lzardsProvider: 'BOGUS_PROVIDER',
+        cumulus_message: {
+          outputs: [
+            {
+              source: '{$.originalPayload}',
+              destination: '{$.payload}',
+            },
+            {
+              source: '{$.backupResults}',
+              destination: '{$.meta.backupStatus}',
+            },
+          ],
+        },
+      };
 
-  it('has the expected backup information', () => {
-    const backupStatus = JSON.parse(new TextDecoder('utf-8').decode(lzardsBackupOutput.Payload)).meta.backupStatus;
-    console.log(`backupStatus: ${JSON.stringify(backupStatus)}`);
-    expect(backupStatus[0].status).toBe('COMPLETED');
-    expect(backupStatus[0].statusCode).toBe(201);
-    expect(backupStatus[0].granuleId).toBe(granuleId);
-    expect(backupStatus[0].provider).toBe(provider);
-    expect(backupStatus[0].createdAt).toBe(tenMinutesAgo);
-    expect(backupStatus[0].collectionId).toBe(encodedConstructCollectionId(collection.name, collection.version));
+      await testSetup({ task_config: taskConfig });
```

Comment:
What's the intention behind moving the `beforeAll` logic to this helper function called inside of an assertion?

I think it works fine. We're doing the same set up and setting `beforeAllFailed` so that seems ok if we call the helper in each test assertion...

Reply:
To make it possible to arg drive the test setup behavior.   We're setting up the same test, just running the lambda with a different configuration.     

`beforeAll` would only run it once, which won't test the assertion, and doing `beforeEach` would either be configured the same, or relying on changing a global which feels Less Good ™️  than indirect behavior changes/ favoring explicit over implicit.

Reply:
Ok, that sounds reasonable.